src/lambda/commands/uploadLambda.ts

```diff
@@ -412,3 +441,113 @@ async function uploadZipBuffer(
         localize('AWS.lambda.upload.done', 'Successfully uploaded Lambda function {0}', lambda.name)
     )
 }
+
+export async function findApplicationJsonFile(startPath: vscode.Uri): Promise<vscode.Uri | undefined> {
+    const isTemplateFile = /template.(yaml|yml|json)$/.test(startPath.fsPath)
+    const parentDir = isTemplateFile ? path.dirname(startPath.fsPath) : startPath.fsPath
+
+    if (isCloud9()) {
+        const found = SystemUtilities.findFile(parentDir, '.application.json')
+        return found ? vscode.Uri.parse(found) : undefined
+    } else {
+        const found = await vscode.workspace.findFiles(
+            new vscode.RelativePattern(parentDir, '**/*.application.json'),
+            undefined,
+            1
+        )
+        return found[0]
+    }
+}
+
+export function getLambdaNamesFromApplicationJson(file: vscode.Uri): string[] | undefined {
+    const appData = JSON.parse(fs.readFileSync(file.fsPath, { encoding: 'utf-8' }).toString())
+    return Object.keys(appData['Functions'])
+}
+
+async function listAllLambdaNames(region: string, path?: vscode.Uri) {
+    const lambdaFunctionNames: DataQuickPickItem<string>[] = []
+    const previousLambdas = getSavedLambdas()
+    const profile = globals.awsContext.getCredentialProfileName()
+    if (previousLambdas && profile && previousLambdas[profile] && previousLambdas[profile][region]) {
+        lambdaFunctionNames.push({
+            label: previousLambdas[profile][region],
+            recentlyUsed: true,
+            data: previousLambdas[profile][region],
+        })
+    }
+    if (path) {
+        const appFile = await findApplicationJsonFile(path)
+        if (appFile) {
+            const namesFromAppFile = getLambdaNamesFromApplicationJson(appFile)
+            if (namesFromAppFile) {
+                lambdaFunctionNames.push(
+                    ...namesFromAppFile.map(n => {
```

Comment:
The items in `.application.json` are mapped to a region. We probably should only show items that match to the user-selected region.